Rules/UseCorrectCasing.cs

```diff
@@ -52,6 +55,16 @@ public override IEnumerable<DiagnosticRecord> AnalyzeScript(Ast ast, string file
                 var fullyqualifiedName = $"{commandInfo.ModuleName}\\{shortName}";
                 var isFullyQualified = commandName.Equals(fullyqualifiedName, StringComparison.OrdinalIgnoreCase);
                 var correctlyCasedCommandName = isFullyQualified ? fullyqualifiedName : shortName;
+                var isWindows = true;
+#if CORECLR
+                isWindows = RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
+#endif
+                if (commandInfo.CommandType == CommandTypes.Application && isWindows)
+                {
+                    // For binaries that could exist on both Windows and Linux like e.g. git we do not want to expand
+                    // git to git.exe to keep the script cross-platform compliant
+                    correctlyCasedCommandName = Path.GetFileNameWithoutExtension(correctlyCasedCommandName);
```

Comment:
This is probably still problematic for users who don't want the extension removed -- I can imagine issues being opened. Is there a way to remember the original command is and conditionally add the extension?

Reply:
Maybe:

```csharp
// Cheapest condition first, may also favour JIT/branch prediction since isWindows is always true or false
if (isWindows
    && commandInfo.CommandType == CommandTypes.Application
    && string.IsNullOrEmpty(Path.GetExtension(commandName)))
{
    correctlyCasedCommandName = Path.GetFileNameWithoutExtension(correctlyCasedCommandName);
}
```

Reply:
Hmm, yes, I added this test case and fixed it by using `!Path.HasExtension(commandName)`.

Reply:
Ah much better API!